docs/source/conf.py

```diff
@@ -292,5 +292,36 @@ def inject_minigalleries(app, what, name, obj, options, lines):
         lines.append("\n")
 
 
+def generate_table():
+
+    import torchvision.models as M
+    from tabulate import tabulate
+    import textwrap
+
+    # TODO: this is ugly af and incorrect. We'll need an automatic way to
+    # retrieve weight enums for each section, or manually list them.
```

Comment:
Any thought on that @datumbox ?

Reply:
Yes you are right. We need a registration mechanism for the models. The new Datasets API has one, so part of the reason I didn't want to invest time creating one is to potentially adopt/extend the one on Datasets. Thoughts?

Reply:
The registration of the dataset is very very basic, it's just a decorator that adds the callable/object to a private dict. It would probably make sense to use something similar for the models / weights. Whether we should be relying on the same utils though is up for discussion - as a first version I'd suggest not to merge things and for the models to have a separate implementation. The code is really basic anyway.

Reply:
I'm happy with what you propose and on the technical details you mentioned. I had a similar simple approach on the [original proposal](https://github.com/datumbox/dapi-model-versioning/blob/main/dapi_lib/models/_api.py#L162) of the Multiweights support but I didn't port it to adopt some solution in common with Datasets. The code doesn't have to be the same but I think the interface can be basic and similar. As discuss offline, the only thing different for models is the fact that there is a hierarchy (Detection, Optical Flow, Classification etc) and this needs to be taken into account because names across modules conflict (for example `resnet50` exists both in Classification and Quantizaztion submodules).